src/modules/legacy/hooks/queries/useCheckLegacyAccount.js

```diff
@@ -3,6 +3,6 @@ import { useLegacy } from '@legacy/hooks/queries';
 export default function useCheckLegacyAccount(address) {
   const { data } = useLegacy({ config: { params: { publicKey: address } } });
   return {
-    isMigrated: data?.data.balance === '0',
+    isMigrated: data ? data.data.balance === '0' : true,
   };
 }
```

Comment:
this can be done 

const { data: { data: legacyData } = {} } = useLegacy({ config: { params: { publicKey: address } } });

so that this 

isMigrated: legacyData ? legacyData?.balance === '0' : true,

Reply:
if the api dosnt provide the legacyData app should stay in loading its wrong logic because in loading state its going to reclaim senario

Reply:
@soroushm It's not going to reclaim in loading state, that's what this fix ensures. Setting isMigrated to true as a fallback ensures that it won't go to reclaim while the data loads

Reply:
@eniolam1000752 Resolved